scripts/core.js

```diff
@@ -91,16 +98,32 @@ var Substitution = (() => {
               }
               return result;
             }
-            else {
-              return word;
-            }
+            return (this.outputType === this.OUTPUT_FUNCTION) ?
+              this.replaceWithFn(this.output, word) :
+              word;
           });
 
-        case this.INPUT_REG_EXP:
-          // necessary according to https://stackoverflow.com/q/1520800
-          this.regExp.lastIndex = 0;
-          return string.replace(this.regExp, unescape(this.output));
       }
+
+    }
+
+    replaceWithFn(userReplaceInput, curStrToReplace) {
```

Comment:
Please add some documentation to this method, like in the others. Maybe some of the comments below could be moved to this method documentation.